libr/debug/p/native/linux/linux_debug.c

```diff
@@ -85,9 +85,8 @@ int linux_handle_signals(RDebug *dbg, int tid) {
 	}
 
 	if (siginfo.si_signo > 0) {
-		// Switch the target thread to the signaled thread to allow access
-		// to its registers later.
-		dbg->tid = tid;
+		// Select the signaled thread to allow access to its info later.
+		r_debug_select (dbg, dbg->pid, tid);
```

Comment:
Are you sure that this is necessary? Seems to work without it and this causes the thread to always return to the main pid after an interrupt.

Reply:
`r_debug_reg_get` will fail if the debugger is targeting a running thread instead of the one returned by `waitpid` when the code below runs:
```
if (dbg->glob_libs || dbg->glob_unlibs) {
	ut64 pc_addr = r_debug_reg_get (dbg, "PC");
	if (b && b->internal) {
```
I realized that setting `dbg->tid` directly is ugly and also using `r_debug_select` will cause other issues. I will just remove the code here then as well and create an issue for that.

Reply:
Some parts of the code call `r_debug_reg_get` without switching to the target thread and will fail in a multi-threaded process.

Reply:
Alrighty. Don't forget to open a new issue for this :)